.github/workflows/publish-to-pypi.yml

```diff
@@ -0,0 +1,187 @@
+        run: >-
+          python -m
+          pip install
+          build
+          --user
+      - name: Build a binary wheel and a source tarball
+        run: >-
+          python -m
+          build
+          --sdist
+          --wheel
+          --outdir dist/
+          .
+      - name: Save built package
+        uses: actions/upload-artifact@v3
+        with:
+          name: dist
+          path: dist/
+          retention-days: 1
+      - name: Publish to Test PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          skip_existing: true
+          password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+          repository_url: https://test.pypi.org/legacy/
+      - name: Wait publishing to TestPyPI
+        uses: juliangruber/sleep-action@v2.0.0
+        with:
+          time: 300s
+  unit-tests-test:
+    name: Running unit tests for TestPyPI
+    needs: build-n-publish-test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ '3.7' , '3.8', '3.9', '3.10', '3.11' ]
+        airflow-version: [ '2.0.2' , '2.2.5', '2.3.4', '2.4.3', '2.5.0' ]
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          rm -rf airflow_clickhouse_plugin
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4.4.0
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+          [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
+          python -m pip install --upgrade pip
+          python -m pip install -r requirements.txt --ignore-installed apache-airflow==${{ matrix.airflow-version }}
+          python -m pip install --force-reinstall --index-url https://test.pypi.org/simple --extra-index-url https://pypi.org/simple airflow-clickhouse-plugin[pandas]==$VERSION
+      - name: Run unit tests
+        run: |
+          python -m unittest discover -s tests/unit
+  integration-tests-test:
+    name: Running integration tests for TestPyPI
+    needs: build-n-publish-test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ '3.7' , '3.8', '3.9', '3.10', '3.11' ]
+        airflow-version: [ '2.0.2' , '2.2.5', '2.3.4', '2.4.3', '2.5.0' ]
+    services:
+      clickhouse:
+        image: yandex/clickhouse-server
+        ports:
+          - 9000/tcp
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          rm -rf airflow_clickhouse_plugin
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4.4.0
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+          [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
```

Comment:
Could you please provide some sample inputs-outputs for this `sed` magic? :)

Reply:
The first sed leaves only the tag:
`a/b/c -> c`
`123/456 -> 456`
`refs/tags/<tag_name> -> <tag_name>`
`refs/tags/v0.11.1 -> v0.11.1`

The second one removes the "v" at the beginning:
`vc -> c`
`v456 -> 456`
`v<tag_name> -> <tag_name>`
`v0.11.1 -> 0.11.1`

Reply:
please add this as a comment to the code, this will help a lot when reading :)